specification/billing/resource-manager/Microsoft.Billing/preview/2020-11-01-preview/billing.json

```diff
@@ -0,0 +1,9330 @@
+        "x-ms-examples": {
+          "AgreementsListByBillingAccount": {
+            "$ref": "./examples/AgreementsListByBillingAccount.json"
+          }
+        },
+        "operationId": "Agreements_ListByBillingAccount",
+        "description": "Lists the agreements for a billing account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountNameParameter"
+          },
+          {
+            "name": "$expand",
+            "description": "May be used to expand the participants.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/AgreementListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/agreements/{agreementName}": {
+      "get": {
+        "tags": [
+          "Agreements"
+        ],
+        "x-ms-examples": {
+          "AgreementByName": {
+            "$ref": "./examples/AgreementByName.json"
+          }
+        },
+        "operationId": "Agreements_Get",
+        "description": "Gets an agreement by ID.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/agreementNameParameter"
+          },
+          {
+            "name": "$expand",
+            "description": "May be used to expand the participants.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/Agreement"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/promotions/{promotionId}": {
+      "put": {
+        "summary": "Activate promotion",
+        "description": "Activate promotion and create resource promotion resource.",
+        "operationId": "Activate_Promotion",
+        "x-ms-examples": {
+          "PromotionActivation": {
+            "$ref": "./examples/PromotionActivation.json"
+          }
+        },
+        "parameters": [
```

Comment:
is there a request body that is needed to create or activate a promotion? Where are things like applied scoped takenn as input?

Reply:
yes let me add that to the e.g.